### ISSUE TITLE
upgrade to PHP 8.0

### ISSUE THREAD

**Issue by megahirt** [sillsdev/web-languageforge#1649]
#1588 must be completed before this

**Comment by megahirt** [comment]
won't do this upgrade - we are now in maintenance mode

**Comment by rmunn** [comment]
I remember looking into PHP 8 years ago and concluding that one of the breaking changes would cause major issues for us. https://www.php.net/manual/en/migration80.incompatible.php says:

> The ability to use [array_key_exists()](https://www.php.net/manual/en/function.array-key-exists.php) with objects has been removed. [isset()](https://www.php.net/manual/en/function.isset.php) or [property_exists()](https://www.php.net/manual/en/function.property-exists.php) may be used instead.

We have 120 uses of `array_key_exists` across 38 files, most of which appear to be used on objects. It's possible that we could switch them over to use `property_exists` instead, but we'd need to do extensive testing to make sure that didn't break anything due to subtly-different behavior. We use lazy properties, and I don't know what the interaction of `property_exists` with lazy properties would do.

So the upgrade to PHP 8.0 would involve quite a bit of work, and I agree with not doing it while in maintenance mode.
